### DISCUSSION TITLE
Open docs tree when open note

### DISCUSSION THREAD

**Discussion by shirobachi** [TriliumNext/Trilium#3572] [Ideas]
![Recording 2023-01-30 at 14 52 58](https://user-images.githubusercontent.com/16509749/215496382-c1f2ec24-cc38-4bed-bcdd-75b91c980f42.gif)

Just open thre tree when go into note.

Also I think that auto close tree (when other opened) would be great

Simon;)

**Comment by zadam** [comment]
> Just open thre tree when go into note.

Hi, unfortunately I don't understand what you mean. When you navigate to a note, the tree will expand to focus on that note.

**Reply by shirobachi** [reply]
Like I would expect that when clicked to template, then on page list would be focused and expanded, RN is only focused

**Reply by zadam** [reply]
I think that would make it hard to click through list of notes. Imagine you have notes like this:

* note 1
* note 2
* note 3

Each of these have a large subtree. You want to click on each of them, so start with note 1, but that expands the large subtree. Clicking on note 2 is then difficult and potentially requires scrolling.

I think it's better to keep the focus and expand separate.

**Comment by shirobachi** [comment]
I will not say that I am not wrong, but see how VS Code expand files:

![Recording 2023-02-10 at 18 48 45 (1)](https://user-images.githubusercontent.com/16509749/218160884-e7a2b5a0-57fe-44d2-9666-8e6cc551ba44.gif)

I actually would prefer to only one tree be open at the time. So in that case `task` common, I want to close when open `docs`.

But what if I would say, let's add it as an optional setting?

**Reply by zadam** [reply]
> But what if I would say, let's add it as an optional setting?

I know it may seem like a good idea, but adding options is very costly. It creates another flow which needs to be tested. Sometimes different options interact between each other, so adding one option actually creates 5 different scenarios. It's better to choose "the best way" and stick with it.

**Reply by shirobachi** [reply]
Got it, thanks!
